scipy/ndimage/tests/test_filters.py

```diff
@@ -787,19 +784,17 @@ def test_correlate1d_complex_input_and_kernel(self, dtype, dtype_output, xp):
         array = xp.asarray([1, 2j, 3, 1 + 4j, 5, 6j], dtype=dtype)
         self._validate_complex(xp, array, kernel, dtype_output)
 
+    @xfail_xp_backends('cupy', reasons=["cupy/cupy#8405",])
+    @skip_xp_backends(np_only=True,
+                      reasons=["output=dtype is numpy-specific"],
+                      exceptions=['cupy'],)
```

Comment:
This is one of the most complex cases. Here is my understanding of what the logic seems to be doing. Let me know if this is correct @ev-br .

- At the module level, we skip for all backends apart from NumPy, CuPy and JAX. CuPy and JAX have some delegation implemented, but other alternative backends do not.
- At the level of this test, `"output=dtype is numpy-specific"` hence we want to skip for JAX as well. But CuPy is okay here (?) so we make an exception for CuPy.
- At the level of this test, we xfail for CuPy due to the linked CuPy issue.

If that is correct, then personally I am "okay" with this syntax, although it would be nice to have something more simple.